GDT.md

```diff
@@ -214,56 +229,52 @@ A more complex example of a GDT is the one used by the stivale2 boot protocol:
+
 ```c
-uint64_t gdt_entries[]; //populate this as you will
+//populate these as you will.
+uint64_t num_gdt_entries;
+uint64_t gdt_entries[];
 
-//packed is required here, otherwise the compiler will insert padding, resulting in the cpu getting bad data.
-__attribute__((packed))
 struct GDTR
 {
     uint16_t limit;
     uint64_t address;
-};
+} __attribute__((packed));
 
-GDTR exampleGdtr = 
+GDTR example_gdtr = 
 {
-    .limit = 5 * sizeof(uint64_t); //assuming we have 5 gdt entries, like the first example setup.
-    .address = (uint64_t)gdtEntries;
+    .limit = num_gdt_entries * sizeof(uint64_t) - 1;
+    .address = (uint64_t)gdt_entries;
 };
 
-void load_and_flush_gdt()
+void load_gdt()
 {
-    /*
-        We use inline assembly to load the gdtr we created above. 
-        the "m" constraint says that its argument must be a memory address (pointer). 
-        The compiler will pass in the address of our exampleGdtr from above to satisfy this.
-        If you're not familiar with gcc's extended inline asm, %0 means the first arg, which is what "m"() represents.
-    */
-    asm("lgdt %0" : : "m"(exampleGdtr));
-
-    /*  A number of assumptions are made in the following code:
-        - Kernel data selector is 0x10, and kernel code is 0x8. Change this for your system if using a different setup.
-        - a call instruction was used to get here, we need the return address to be sitting on the top of the stack.
-            - most compilers do this, so not much to worry about. Just be aware of it if calling from assembly.
-
-        The first part of the code below updates the data segment registers with our new data segment.
-        The second part turns the existing return address into a far return, to reload the CS register.
-    */
-    asm("\
-            mov $0x10, %ax \n\
-            mov %ax, %ds \n\
-            mov %ax, %es \n\
-            mov %ax, %fs \n\
-            mov %ax, %gs \n\
-            mov %ax, %ss \n\
-            \n\
-            pop %rdi \n\
-            push $0x8 \n\
-            push %rdi \n\
-            lretq \n\
-        ");
-
-    //at this point we've successfully installed our new gdt, and reloaded the segment registers.
+    asm("lgdt %0" : : "m"(example_gdtr));
 }
 ```
+
+If you're not familiar with inline assembly, check the appendix on using inline ssembly in C. The short of it is we use the "m" constraint to tell the compiler that `example_gdtr` is a memory address. The `lgdt` instruction loads the new GDT, and all that's left is to reload the current selectors, since they're using cached information from the previous GDT.
+
+```c
+void flush_gdt()
+{
+    asm volatile("\
+                mov $0x10, %ax \n\
+                mov %ax, %ds \n\
+                mov %ax, %es \n\
+                mov %ax, %fs \n\
+                mov %ax, %gs \n\
+                mov %ax, %ss \n\
+                \n\
```

Comment:
just double checking: is this empty line intentional? if yes just ignore leave as it is. 

Why the extra \ at the end of each line?

Reply:
It tells the compiler that the string doesn't end at the end of the line.
Empty line is for clarity.